class/class.midtrans-gateway-notif-handler.php

```diff
@@ -290,4 +287,37 @@ public function validateRefundNotif( $midtrans_notification ) {
     return $refund_request;
   }
 
+  /**
+   * Process subscription transaction if contains one of those
+   * 
+   * @param [Object] $midtrans_notification Object representation of Midtrans JSON notification
+   * @param WC_Order $order 
+   * @return void
+   */
+  public function validateSubscriptionTransaction( $midtrans_notification, $order ) {
```

Comment:
Idea:
Can if-else branching on this function be simplified? for example by breaking down "do-nothing scenario, or exit-early". 
For example instead of
```php
      // Don't process if contains renewal and the order status is pending
      if ( !wcs_order_contains_renewal( $midtrans_notification->order_id) || $order->get_status() != 'pending' ) {
          // perform process
          if (....){

          } else (){

          }
       }
```

it become
```php
// if not subscription and  wc status pending, don't process
if( !wcs_order_contains_renewal( $midtrans_notification->order_id) && $order->get_status() == 'pending' ){
  return false;
}

// continue process

// another dont-process condition
if (... ){ 
   return false;
}
// continue process
```
Try it, you will have less `else` branch and this is easier to read and understand in the future.

Or if its to complex to refactor, then you can leave it as it is for now.

Reply:
done on https://github.com/veritrans/SNAP-Woocommerce/pull/39/commits/b766b50580be825d654c7779858dc1e48583d03b